students/hangu/users/views.py

```diff
@@ -1,3 +1,42 @@
-from django.shortcuts import render
+import json
+import re
 
-# Create your views here.
+from django.http import JsonResponse
+from django.views import View
+
+from .models import User
+
+class SignupView(View):
+    def post(self, request):
+        try:
+            user_data = json.loads(request.body)
+            
+            user_name               = user_data['user_name']
+            user_email              = user_data["user_email"]
+            password                = user_data["password"]
+            phone_number            = user_data["phone_number"]
+            email_validation        = re.compile("^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$")
+            password_validation     = re.compile("^.*(?=^.{8,}$)(?=.*\d)(?=.*[a-zA-Z])(?=.*[!@#$%*^&+=]).*$")
+            phone_number_validation = re.compile("^01([0|1|6|7|8|9])-?([0-9]{3,4})-?([0-9]{4})$")
+            
+            if user_email == "" or password == "":
+                return JsonResponse({"message":"이메일 또는 비밀번호를 입력해주세요"}, status=400)
+                
+            if not email_validation.match(user_email):
+                return JsonResponse({"message":"이메일양식이 잘못되었습니다"}, status=400)
+
+            if not password_validation.match(password):
+                return JsonResponse({"message":"비밀번호는 8자리 이상, 문자, 숫자, 특수문자의 복합이어야합니다. "}, status=400)
+            
+            if not phone_number_validation.match(phone_number):
+                return JsonResponse({"message":"핸드폰번호가 잘못되었습니다"}, status=400)
+
+            user_email = User.objects.filter(user_email = user_email).first()
+            if user_email is None:
```

Comment:
사용자의 존재 유무만 판단하는 경우에 first() 메소드로 객체를 가져오는 것 보다는 .exists() 함수로 True 혹은 False만 가져오는 것이 성능상 이득을 볼 수 있습니다. 찾아보고 적용 부탁드립니다-!

```suggestion
            user_email = User.objects.filter(user_email = user_email).first()
            if user_email is None:
```